diff --git company-mlton.el company-mlton.el
index 20063cf..e82d40d 100644
--- ./lisp/company-mlton/company-mlton.el
+++ ./lisp/company-mlton/company-mlton.el
@@ -75,7 +75,7 @@ Only handles a small subset of `rx' sexp forms."
 (defconst company-mlton--sml-alphanum-rx
   `(char "A-Z" "a-z" "0-9" "'" "_"))
 (defconst company-mlton--sml-alphanum-id-rx
-  `(: (char "A-Z" "a-z") (* ,company-mlton--sml-alphanum-rx)))
+  `(: (char "A-Z" "a-z" "_") (* ,company-mlton--sml-alphanum-rx)))
 (defconst company-mlton--sml-sym-rx
   `(char "!" "%" "&" "$" "#" "+" "-"
          "/" ":" "<" "=" ">" "?" "@"
@@ -186,9 +186,19 @@ Otherwise, return 'nil."
   '("eqtype" "functor" "include" "sharing" "sig"
     "signature" "struct" "structure" "where")
   "The list of Standard ML keywords for the modules language.")
+(defconst company-mlton-keyword--sml-keywords-smlsharp
+  '("all" "asc" "begin" "by" "commit" "default" "delete" "desc"
+    "distinct" "from" "insert" "into" "order" "rollback" "select"
+    "update" "use" "values"
+    "_NULL" "_attribute" "_builtin" "_ffiapply" "_foreach"
+    "_import" "_interface" "_join" "_json" "_jsoncase" "_polyrec"
+    "_require" "_sizeof" "_sql" "_sqleval" "_sqlexec" "_sqlserver"
+    "_typeof" "_use")
+  "The list of SML# keywords.")
 (defconst company-mlton-keyword--sml-keywords
   (sort (append company-mlton-keyword--sml-keywords-core
-                company-mlton-keyword--sml-keywords-modules)
+                company-mlton-keyword--sml-keywords-modules
+                company-mlton-keyword--sml-keywords-smlsharp)
         'string<)
   "The list of Standard ML keywords.")
 
