diff --git defwithdoc defwithdoc
index 6c8fa04..fea7afb 100644
--- ./snippets/python-mode/defwithdoc
+++ ./snippets/python-mode/defwithdoc
@@ -4,12 +4,16 @@
 #contributor : MATSUBARA Naoya
 # --
 def ${1:function}(${2:args}):
-    """${3:short description}
-
-    $0
+    """
+    ${3:short description}
 
     ${4:Parameters
     ----------
     ${5:Please press the tab key twice.$(replace-regexp-in-string
                                            ".*" "parameters" yas-text)}
-    }"""
\ No newline at end of file
+    }
+
+    Returns
+    -------
+    ${6:returns}
+    """
\ No newline at end of file
