--- ./elpa/sml-mode-6.9/sml-mode.el.orig	2019-03-13 19:50:04.405119600 +0900
+++ ./elpa/sml-mode-6.9/sml-mode.el	2019-03-13 19:55:29.017351700 +0900
@@ -255,7 +255,16 @@
                  "infixr" "let" "local" "nonfix" "o" "of" "op" "open" "orelse"
                  "overload" "raise" "rec" "sharing" "sig" "signature"
                  "struct" "structure" "then" "type" "val" "where" "while"
-                 "with" "withtype"))
+                 "with" "withtype"
+                 ; SML# extentions
+                 "all" "asc" "begin" "by" "commit" "default" "delete" "desc"
+                 "distinct" "from" "insert" "into" "order" "rollback" "select"
+                 "update" "use" "values"
+                 "_NULL" "_attribute" "_builtin" "_ffiapply" "_foreach"
+                 "_import" "_interface" "_join" "_json" "_jsoncase" "_polyrec"
+                 "_require" "_sizeof" "_sql" "_sqleval" "_sqlexec" "_sqlserver"
+                 "_typeof" "_use"
+                 ))
   "A regexp that matches any and all keywords of SML.")
 
 (eval-and-compile
